### PR TITLE
First step; ruby support

### PR DESCRIPTION
**Pull request by justincan**
Still pretty early, not ready for real use.

### CODE REVIEW

**Comment by elopio** [comment]
Thanks @justincan. I'm okay landing this to get started, but it needs tests.
To start, you can look at integration_tests and add a build for a basic ruby hello world.
Feel free to let me know if you need a hand.

**Comment by sergiusens** [comment]
any update on this

**Comment by justincan** [comment]
I have some improvements and tests that I can add to this PR. Still more to make it solid though.

File: snapcraft/plugins/ruby.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    self.stage_packages.append('ruby')
+    if self.options.will_install_gems:
+      self.stage_packages.extend(build_packages)
+
+    self.build_packages.extend(build_packages)
+
+  def env(self, root):
+    ruby_paths = self.run_output(['ruby', '-e', 'puts $:']).split(os.linesep)
+    ruby_paths = ':'.join(['$SNAP' + x for x in ruby_paths])
```
Comment (kyrofa) [inline snapcraft/plugins/ruby.py]: This seems like it will break terribly if one is using RVM, no?

File: snapcraft/plugins/ruby.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    schema['properties']['gems'] = {
+      'type': 'array',
+      'minitems': 1,
+      'uniqueItems': True,
+      'items': {
+        'type': 'string',
+      },
+      'default': [],
+    }
+    schema['properties']['will-install-gems'] = {
```
Comment (kyrofa) [inline snapcraft/plugins/ruby.py]: I don't really understand what this option does. It has "gems" in the name, but really just seems to optionally add build packages to stage packages. What does that have to do with gems?

**Comment by kyrofa** [comment]
Closing due to inactivity. Please reopen when you're ready to continue.

**Comment by elopio** [comment]
For the record, here's our experiment with a ruby snap: https://github.com/travis-ci/travis.rb/pull/515

**Pull request closed without merging**